### ISSUE TITLE
vstest.console.exe for UWP test project fails with error "Failed to initialize client proxy: could not connect to test process"

### ISSUE THREAD

**Issue by DanieleScipioni** [Microsoft/vstest#1675]
## Description 
I'm trying to automate tests for an UWP project using vstest.console.exe, when I run the command it ends with message `Test Run Aborted`. I expect to find a .trx file with the outcome of the tests.

Currently I'm using Visual Studio 15.7.4 and Microsoft (R) Test Execution Command Line Tool Version 15.7.2 on Windows 10.0.16299.492

## Steps to reproduce
I run the following command:

```vstest.console.exe /Platform:x86 /Framework:FrameworkUap10 /InIsolation /logger:trx;LogFileName=project.test.trx Project.Test\AppPackages\Project.Test_1.0.0.0_x86_Test\Project.Test_1.0.0.0_x86.appx```

and I get the following error

```Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.```

This is the complete output:

```
Microsoft (R) Test Execution Command Line Tool Version 15.7.2
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Attempting to deploy to target 'Local Machine'.
Deployment operation succeeded, package moniker: 'Project.Test_1.0.0.0_x86__b5bx2qemknzx6'.
Attempting to launch app with app user model Id 'Project.Test_b5bx2qemknzx6!vstest.executionengine.universal.App' and arguments '--port 53217 --endpoint 127.0.0.1:053217 --role client --parentprocessid 14380 --telemetryoptedin false'.
Successfully launched process Id '5464'.
Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
Closing app with package full name 'Project.Test_1.0.0.0_x86__b5bx2qemknzx6'.

Test Run Aborted.
```

When I run the command, the test window is opened for a moment just after log line `Successfully launched process Id '5464'.` and then closed. After few seconds the exception is written.

**Comment by kiran-mahadev** [comment]
I'm running into the same issue as well when executing the VSTest step from TFS. 

Everything works fine locally (on my laptop) but the same step setup as part of a TFS build task fails with the error - Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.

**Comment by DanieleScipioni** [comment]
@mayankbansal018: 2 answers.

>  you can use ".appxrecipe" (**recommended**) as input to CLI.

thanks for the hint, please note that the full command line is welcome in order to avoid mistakes following the input (same reasons listed to create a new issue).

To use ".appxrecipe" I run this command line:

`vstest.console.exe /Platform:x86 /Framework:FrameworkUap10 /InIsolation /logger:trx;LogFileName=project.test.trx Project.Test\bin\x86\Release\Project.Test.build.appxrecipe`

this is the output

```
Microsoft (R) Test Execution Command Line Tool Version 15.7.2
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Attempting to deploy to target 'Local Machine'.
Updating the layout...

Deployment complete (0:00:00.159). Full package name: "Project.Test_1.0.0.0_x86__b5bx2qemknzx6"

Deployment operation succeeded, package moniker: 'Project.Test_1.0.0.0_x86__b5bx2qemknzx6'.
Attempting to launch app with app user model Id 'Project.Test_b5bx2qemknzx6!vstest.executionengine.universal.App' and arguments '--port 51752 --endpoint 127.0.0.1:051752 --role client --parentprocessid 1568 --telemetryoptedin false'.
Successfully launched process Id '3604'.
Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
Closing app with package full name 'Project.Test_1.0.0.0_x86__b5bx2qemknzx6'.

Test Run Aborted.
```

so I get exactly the same error.

----------

> If you need to make things run via ".appx", please make sure you have uninstalled the "Test App", this app could have gotten installed while running the test from IDE, & try again.

Yes, I prefer to run test via `".appx"`. I'm sure that "Test App" is uninstalled when I run the command, moreover if "Test App" were installed the error would be

`DEP0820: The appx file 'C:\Repos\bitbucket\UWP\Project.Test\AppPackages\Project.Test_1.0.0.0_x86_Test\Project.Test_1.0.0.0_x86.appx' failed to install. [0x80073CFB]`

**Comment by mayankbansal018** [comment]
@DanieleScipioni , few follow up questions: 

- Do you see the same error, if you run the test via VS IDE?
- When you run via CLI do you see the TestApp being launched, or not. I can see from your above logs that it says the the App did launch, just wanted a confirmation.
- Can you enable diagnostics logging via  "_vstest.console.exe project.Test\bin\x86\Release\Project.Test.build.appxrecipe /diag:out.txt"_. You should see path for two log files displayed on console, can you please share both the files.
- From the above logs it says that could not connect to test process, can you make sure that your "Package.appxmanifest" has both "internetClientServer" & "privateNetworkClientServer" capabilities. 
- Also can please verify whether you are able to run test for a blank UWP UT app?

**Comment by DanieleScipioni** [comment]
> Do you see the same error, if you run the test via VS IDE?

No, via VS IDE test are executed correctly.

> When you run via CLI do you see the TestApp being launched, or not. I can see from your above logs that it says the the App did launch, just wanted a confirmation.

Yes, the TestApp is launched for a moment just after log line `Successfully launched process Id 'XXXX'.` and then it closes. It seems that the TestApp runs just for tenths of a second. I used the Task Manager to look for process with the given id without success.

> Can you enable diagnostics logging via "vstest.console.exe project.Test\bin\x86\Release\Project.Test.build.appxrecipe /diag:out.txt". You should see path for two log files displayed on console, can you please share both the files.

I used the command
`vstest.console.exe /Platform:x86 /Framework:FrameworkUap10 /InIsolation /logger:trx;LogFileName=project.test.trx /diag:out.txt Project.Test\bin\x86\Debug\Project.Test.build.appxrecipe`

this is the ouput

```
Microsoft (R) Test Execution Command Line Tool Version 15.7.2
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Logging Vstest Diagnostics in file: C:\Repos\UWP\out.txt
Attempting to deploy to target 'Local Machine'.
Creating a new clean layout...

Copying files: Total 22 mb to layout...

Checking whether required frameworks are installed...

Registering the application to run from layout...

Deployment complete (0:01:30.702). Full package name: "Project.Test_1.0.0.0_x86__b5bx2qemknzx6"

Deployment operation succeeded, package moniker: 'Project.Test_1.0.0.0_x86__b5bx2qemknzx6'.
Attempting to launch app with app user model Id 'Project.Test_b5bx2qemknzx6!vstest.executionengine.universal.App' and arguments '--port 53715 --endpoint 127.0.0.1:053715 --role client --parentprocessid 18672 --diag "C:\Repos\bitbucket\UWP\out.host.18-07-20_19-08-35_64639_6.txt" --telemetryoptedin false'.
Successfully launched process Id '14800'.
Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
Closing app with package full name 'WuerthPhoenix.Speedy.DataAccess.Test_1.0.0.0_x86__b5bx2qemknzx6'.

Test Run Aborted.
```
I get only one output file [out.txt](https://github.com/Microsoft/vstest/files/2214850/out.txt) attached.


> From the above logs it says that could not connect to test process, can you make sure that your "Package.appxmanifest" has both "internetClientServer" & "privateNetworkClientServer" capabilities.

I tried with the capabilities you suggest without success.

> Also can please verify whether you are able to run test for a blank UWP UT app?

With the black UWP UT app I have no problem.

**Comment by mayankbansal018** [comment]
@DanieleScipioni , since you are only getting one log file, it means that TestApp is failing immediately after it is launched. We might need to have a session to figure this, can you drop a mail to devops_tools@microsoft.com

**Comment by DanieleScipioni** [comment]
Now I'm able to run tests with CLI, I made the following changes in Project.Test.csproj file:

- changed references to test framework libraries:
```xml
<SDKReference Include="MSTestFramework.Universal, Version=$(UnitTestPlatformVersion)" />
```
replaced with 
```xml
<PackageReference Include="MSTest.TestAdapter">
  <Version>1.3.2</Version>
</PackageReference>
<PackageReference Include="MSTest.TestFramework">
  <Version>1.3.2</Version>
</PackageReference>
```
- added
```xml
<PropertyGroup>
  <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
</PropertyGroup>
```

**Comment by mayankbansal018** [comment]
@DanieleScipioni , good know that you got the issue fixed, can you please try one more thing i.e. remove 

```xml
<SDKReference Include="MSTestFramework.Universal, Version=$(UnitTestPlatformVersion)" />
```
& add 
replaced with 
```xml
<PackageReference Include="MSTest.TestAdapter">
  <Version>1.3.1</Version>
</PackageReference>
<PackageReference Include="MSTest.TestFramework">
  <Version>1.3.1</Version>
</PackageReference>
```

Just change the adaper version to 1.3.1 or lower

**Comment by DanieleScipioni** [comment]
@mayankbansal018 do not understand what you want me to do. I already removed `SDKReference` and I am already using version 1.3.1.
Maybe my comment was unclear? I repeat here the changes i did to make the test project working:

- I removed SDKReference to `MSTestFramework.Universal`
- I added PackageReference to `MSTest.TestAdapter` and `MSTest.TestFramework` (both with version 1.3.1)
- I added `<RestoreProjectStyle>PackageReference</RestoreProjectStyle>`
